lms/djangoapps/certificates/views/support.py

```diff
@@ -0,0 +1,185 @@
+
+    Arguments:
+        request (HttpRequest): The request object.
+
+    Returns:
+        JsonResponse
+
+    Example Usage:
+        GET /certificates/search?query=bob@example.com
+
+        Response: 200 OK
+        Content-Type: application/json
+        [
+            {
+                "username": "bob",
+                "course_key": "edX/DemoX/Demo_Course",
+                "type": "verified",
+                "status": "downloadable",
+                "download_url": "http://www.example.com/cert.pdf",
+                "grade": "0.98",
+                "created": 2015-07-31T00:00:00Z,
+                "modified": 2015-07-31T00:00:00Z
+            }
+        ]
+
+    """
+    query = request.GET.get("query")
+    if not query:
+        return JsonResponse([])
+
+    try:
+        user = User.objects.get(Q(email=query) | Q(username=query))
+    except User.DoesNotExist:
+        return JsonResponse([])
+
+    certificates = api.get_certificates_for_user(user.username)
+    for cert in certificates:
+        cert["course_key"] = unicode(cert["course_key"])
+        cert["created"] = cert["created"].isoformat()
+        cert["modified"] = cert["modified"].isoformat()
+
+    return JsonResponse(certificates)
+
+
+def _validate_regen_post_params(params):
+    """
+    Validate request POST parameters to the regenerate certificates end-point.
+
+    Arguments:
+        params (QueryDict): Request parameters.
+
+    Returns: tuple of (dict, HttpResponse)
+
+    """
+    # Validate the username
+    try:
+        username = params.get("username")
+        user = User.objects.get(username=username)
+    except User.DoesNotExist:
+        msg = _("User {username} does not exist").format(username=username)
+        return None, HttpResponseBadRequest(msg)
+
+    # Validate the course key
+    try:
+        course_key = CourseKey.from_string(params.get("course_key"))
+    except InvalidKeyError:
+        msg = _("{course_key} is not a valid course key").format(course_key=params.get("course_key"))
+        return None, HttpResponseBadRequest(msg)
+
+    return {"user": user, "course_key": course_key}, None
+
+
+@require_POST
+@require_certificate_permission
+def regenerate_certificate_for_user(request):
+    """
+    Regenerate certificates for a user.
+
+    This is meant to be used by support staff through the UI in lms/djangoapps/support
+
+    Arguments:
+        request (HttpRequest): The request object
+
+    Returns:
+        HttpResponse
+
+    Example Usage:
+
+        POST /certificates/regenerate
+            * username: "bob"
+            * course_key: "edX/DemoX/Demo_Course"
+
+        Response: 200 OK
+
+    """
+    # Check the POST parameters, returning a 400 response if they're not valid.
+    params, response = _validate_regen_post_params(request.POST)
+    if response is not None:
+        return response
+
+    # Check that the course exists
+    course = modulestore().get_course(params["course_key"])
```

Comment:
Could you use the CourseOverview cache here instead of hitting the modulestore directly?